packages/aztec.js/src/proof/swap/verifier.js

```diff
@@ -24,20 +24,34 @@ class SwapVerifier extends Verifier {
         challengeResponse.appendBN(this.proof.sender.slice(2));
         challengeResponse.data = [...challengeResponse.data, ...rollingHash.data];
 
+        let reducer = bn128.zeroBnRed;
         this.data.forEach((note, i) => {
+            let aBar;
             let kBar;
-            const { aBar, gamma, sigma } = note;
+            let c;
+            const { gamma, sigma } = note;
 
-            if (i <= 1) {
+            if (i === 0) {
```

Comment:
Raising that we could check only if i == 1 and i == 2, since we only care about the pairing on output notes.